src/formatPkg.js

```diff
@@ -222,6 +239,65 @@ function getHomePage(homepage, repository) {
   return null;
 }
 
+/**
+ * Get info from urls like this: (has multiple packages in one repo, like babel does)
+ *  https://github.com/babel/babel/tree/master/packages/babel
+ *  https://gitlab.com/user/repo/tree/master/packages/project1
+ *  https://bitbucket.org/user/repo/src/ae8df4cd0e809a789e3f96fd114075191c0d5c8b/packages/project1/
+ *
+ * This function is like getGitHubRepoInfo (above), but support github, gitlab and bitbucket.
+ */
+function getRepositoryInfoFromHttpUrl(repository) {
+  const result = repository.match(
+    /^https?:\/\/(?:www\.)?((?:github|gitlab|bitbucket)).((?:com|org))\/([^/]+)\/([^/]+)(\/.+)?$/
+  );
+
+  if (!result || result.length < 4) {
+    return null;
+  }
+
+  return {
+    host: `${result[1]}.${result[2]}`,
```

Comment:
Here (and above as well) it would be prettier if we destructured `result`

Reply:
You mean like this:

```js
  const [, domain, domainTld, user, project, path = ''] = result;
  
  return {
    host: `${domain}.${domainTld}`,
    user,
    project,
    path,
  };
```

Reply:
yep, that seems nicer to me

Reply:
> and above as well

Where? do you mean in another function? which one?

Reply:
I meant in `getGithubInfo`, to which you referred in the comment